autotst/job/job.py

```diff
@@ -42,21 +43,54 @@ class Job():
 
     def __init__(
             self,
-            reaction=None,
+            species=None,
+            reactions=None,
+            load_species_from_reactions = False,
             calculator=None, # An AutoTST Gaussian calculator with proper directory settings
             conformer_calculator = None, # an ASE Calculator object
             partition="general", # The partition to run calculations on
             directory = None, # where to save your files 
             scratch = None # a directory for temporary files generated by calculators
             ):
 
-        self.reaction = reaction
-        if isinstance(reaction, Reaction):
-            self.label = self.reaction.label
-        elif reaction is None:
-            self.label = None
+        
+        if species is not None:
+            if isinstance(species,list):
+                for s in species:
+                    assert isinstance(s,Species),"species {} provided was not an AutoTST species object".format(s)
+                self.species = species # sets self.species to list of AutoTST species
+            elif isinstance(species,Species):
+                self.species = [species] # sets self.species to list of a single AutoTST species 
+            else:
+                assert False, "Species must be an AutoTST species or a list or AutoTST species" 
+        else:
+            self.species = None
+
+        if reactions is not None:
+            if isinstance(reactions,list):
+                for r in reactions:
+                    assert isinstance(r,Reaction),"reaction {} provided was not an AutoTST reaction object".format(r)
+                self.reactions = reactions # sets self.reactions to list of AutoTST reactions
+                self.label = [reaction.label for reaction in self.reactions]
+            elif isinstance(reactions,Reaction):
+                self.reactions = [reactions] # sets self.reactions to list of a single AutoTST reaction 
+                self.label = reactions.label
+            else:
+                assert False, "Reactions must be an AutoTST reaction of list of AutoTST reactions"
         else:
-            assert False, "Reaction provided was not a reaction object"
+            self.reactions = None
+            self.label = None
+
+        if load_species_from_reactions is True:
+            reaction_species = self.get_species_from_reactions()
+            if self.species is None:
+                self.species = reaction_species
+            else:
+                species = self.species
+                for s in reaction_species:
+                    if s not in species:
```

Comment:
I'm not sure if this would work because you create multiple individual `Species` objects that don't point to the same data. This _probably_ will just make a list with multiple duplicates...